### ISSUE TITLE
No data when DSMR is enabled

### ISSUE THREAD

**Issue by EvertJob** [heyajohnny/afvalinfo#94]
Hi,

I'm having a strange issue.

This is my configuration.yaml file:
```
sensor:                       
    - platform: afvalinfo     
      resources:              
        - gft            
        - papier         
        - pbd        
        - trash_type_today
        - trash_type_tomorrow
      city: almere           
      postcode: f       
      streetnumber: f       
      dateformat: '%d-%m-%Y'
      locale: 'nl'           
      timespanindays: 365    
                             
sensor:                  
    - platform: dsmr_reader 
                            
binary_sensor:              
  - platform: meteoalarm   
    country: 'NL'        
    province: 'Flevoland'
```

With this configuration I can see DSMR data and the data from metroalarm but no data from the afvalinfo plugin. When I remove the DSMR Reader the data from afvalinfo is working as expected.

It doesn't matter which combination is used, as long DSMR and Afvalinfo are in the same configuration one of the two stops working.

I cannot figure out what the issue is, the logs are not giving any data about it so I'm going dark in here.

Are the more people with the same issue, or people who got it working and are willing to share that part of there config?

**Comment by heyajohnny** [comment]
I can't even get 'dsmr_reader' to work :(
`2020-05-31 09:07:29 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of dsmr_reader. Setup failed for dependencies: mqtt`

Maybe it has something to do with https://github.com/home-assistant/core/issues/35879

**Comment by EvertJob** [comment]
Did you enable MQTT?

**Comment by heyajohnny** [comment]
> Did you enable MQTT?

Nope, I didn't. After enabling it, it seems to work just fine...
On the other hand, I don't have a MQTT broker, I just enabled mqtt in the configuration.yaml 

![image](https://user-images.githubusercontent.com/20553716/83360295-83c3a580-a380-11ea-96ea-8808c804e9d2.png)

I'm sorry, but I can't help you any further. You can keep this issue open if you like. Maybe someone else can help you out.

**Comment by EvertJob** [comment]
You are not receiving any data, correct? 

Maybe I need to open an issue at Home-Assistant. For now I started a new Docker Container for this plug-in only. It rocks!

**Comment by heyajohnny** [comment]
> You are not receiving any data, correct?

I don't receive any DSMR data, but that's correct, since I don't have any any DSMR compatible device / mqtt broker. The behaviour is as expected with this empty dsmr_reader at my side. So, basically what I'm saying is that I'm not the guy who can test it further than this :(

**Comment by EvertJob** [comment]
Ahh, oke.

Not a very big issue as said before, just started a new container for this plug-in only. Got an Telegram message true Node-Red :)

**Comment by yniezink** [comment]
Hi, @EvertJob your HA config is wrong. 

`sensor:                       
    - platform: afvalinfo     
      resources:              
        - gft            
        - papier         
        - pbd        
        - trash_type_today
        - trash_type_tomorrow
      city: almere           
      postcode: f       
      streetnumber: f       
      dateformat: '%d-%m-%Y'
      locale: 'nl'           
      timespanindays: 365
                       
    - platform: dsmr_reader `

Try it like above. You are only allowed to setup "sensor" once. You do it twice. Maybe HA then will only use the last one you do

**Comment by yniezink** [comment]
@EvertJob Well since the new version you have to use location instead of city. Did you change that?

I have dmsr_reader and I use afvalinfo and both are working great.

**Comment by EvertJob** [comment]
> @EvertJob Well since the new version you have to use location instead of city. Did you change that?
> 
> I have dmsr_reader and I use afvalinfo and both are working great.
> 
> ```
> sensor:                       
>   - platform: afvalinfo     
>     resources:              
>       - gft            
>       - papier         
>       - pbd        
>       - trash_type_today
>       - trash_type_tomorrow
>     location: almere           
>     postcode: f       
>     streetnumber: f       
>     dateformat: '%d-%m-%Y'
>     locale: 'nl'           
>     timespanindays: 365                 
>   - platform: dsmr_reader
> ```

Just tried, still no data... :(

**Comment by EvertJob** [comment]
Fixed in 0.111.4... Don't know what causes the issue.

**Comment by heyajohnny** [comment]
Good to hear it's fixed! :)
